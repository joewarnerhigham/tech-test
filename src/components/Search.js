import React, {useState} from 'react'
import '../styles/Search.css';
import getImages from '../requests/getImages';


const Search = ( {setSearchResults }) => {
    const [value, setValue] = useState();

    const handleSubmit =(event) =>{
        event.preventDefault()
        getImages(value)
    }
    return (
        <div className='container'>
            <img className='logo'
                 src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg" 
                 alt='nasa-logo'
                 data-testid='logo'
            />
            <form className='search-form' onSubmit={handleSubmit}>
                <input className='search-bar'
                    type='text' 
                    value={value}
                    onChange={(e) => setValue(e.target.value)} 
                    placeholder='Search'
                    data-testid='search-bar'
                />
                <button className='search-button' type='submit'>Go!</button>
            </form>
        </div>
    )
}

export default Search
