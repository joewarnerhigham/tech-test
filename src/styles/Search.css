body{
    display: flex;
    justify-content: center;
}

.container{
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.logo{
    width: 200px;
}

.search-bar{
    border-radius: 4px;
    text-align: center;
    border: rgb(33, 68, 62) solid 2px;
}


