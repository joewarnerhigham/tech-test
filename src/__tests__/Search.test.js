import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Search from '../components/Search';

describe('Search bar', () => {

    afterEach(cleanup)

    it('should have a logo', () => {
        const {getByTestId} = render(
            <Search />
        )
    
    expect(getByTestId('logo'));  
    });

    it('should have search-bar with placeholder', () => {
        const {getByTestId} = render (
            <Search /> 
        )
    
    expect(getByTestId('search-bar')).toHaveClass('search-bar');
    });
});
