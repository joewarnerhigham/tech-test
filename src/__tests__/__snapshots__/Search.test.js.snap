// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt="nasa-logo"
      class="logo"
      data-testid="logo"
      src="https://www.seekpng.com/png/full/393-3935156_nasa-app-app-logo-png-download-nasa-logo.png"
    />
    <form
      class="search-form"
    >
      <input
        class="search-bar"
        data-testid="search-bar"
        placeholder="Search"
        type="text"
      />
      <button
        class="search-button"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
